docs/rules/object-curly-newline.md

```diff
@@ -19,6 +19,7 @@ Or an object option:
 
 * `"multiline": true` (default) requires line breaks if there are line breaks inside properties or between properties
 * `"minProperties"` requires line breaks if the number of properties is more than the given integer
+* `"consistent"` requires that either both curly braces, or neither, directly enclose newlines
```

Comment:
It's not clear to me how this option interacts with the other rule options. For example, if I have `multiline: true`, the `consistent` option is redundant, because `multiline: true` already implies that either curly braces will be required on both braces or neither.

Additionally, it looks like the `minProperties` option always either requires linebreaks or disallows them, contrary to what the documentation says. So I don't think it makes sense to use this option with `minProperties`. We might want to make this option mutually exclusive with some of the other options, to prevent the user from creating conflicting configs.